pyccel/codegen/printing/cwrappercode.py

```diff
@@ -104,6 +104,9 @@ def get_PyArgParseType(self, used_names, variable):
             These handle casting collect_var to variable if necessary
         """
 
+        if isinstance(variable, FunctionAddress):
+            return variable, []
```

Comment:
Does this not give an incompatible type error when you try to compile?

Reply:
it doesn't, why do you suspect that ?

Reply:
The second argument of this return handles casting the variable collected from the argument parsing and translating it to a type that makes sense in pure C. So in this case it should translate a `PyObject*` to a function pointer, e.g. `int(*f)(int)`

I think you decided to leave this translation for another PR but it would be nice if we could raise an error when trying to use the epyccelized function as if you pass a `PyObject*` as a function pointer I suspect that you will have either a compiler problem, or a segmentation fault. You will know more about this than me I think.
In order to raise a python error instead you need to use `PyErr_SetString` and return Nil(). You can search for `PyErr_SetString` in `codegen/printing/cwrappercode.py` for an example of how to raise errors

Reply:
`        if isinstance(variable, FunctionAddress):
            body = [PyErr_SetString('PyExc_NotImplementedError', '"can not pass a function as argument"'), Return([Nil()])]
            return variable, body
`
do you mean something like this?

Reply:
Yes exactly that looks perfect :+1: